### ISSUE TITLE
Running MxNet.jl on AWS GPU instance

### ISSUE THREAD

**Issue by Andy-P** [dmlc/MXNet.jl#43]
I was wondering whether anyone had succeeded in running MXNet.jl using an AWS multi-GPU instances (g2.8xlarge). There is an upcoming Julia Tokyo meet up (12/19) at which I was planning to do a presentation on MxNet.jl. I thought it would be really cool if I could show off the GPU capabilities, but many people don't have GPUs. Being able to show a working example of Julia's "Deep-learning-in-the-cloud-using-GPUs" might generate a lot of interest. 

If no one has tried it yet, I might give it a go. Any advice, tips, or links to helpful scripts would be greatly appreciated.

Andre

**Comment by pluskid** [comment]
I personally have no experience running on AWS. But if there is no issue with Julia on AWS, I guess it should be fine, as it seems people tried libmxnet on AWS. @mli might be more experienced here. The only thing that I heard is that AWS opus are quite old, so I'm not sure the latest cuDNN and stuff runs on them or not.

**Comment by mli** [comment]
here is a public ami with cuda/cudnn installed

https://console.aws.amazon.com/ec2/v2/home?region=us-east-1#LaunchInstanceWizard:ami=ami-12fd8178

see document http://mxnet.readthedocs.org/en/latest/aws.html

it should be easy to build mxnet.jl

one g2.8x (4 gpus, 2 physical cards) is roughly equal to one k40

**Comment by tqchen** [comment]
This is nice @Andy-P  If you build mxnet with S3 support(need to change one line in config.mk) you will be able to put data on S3 and directly point data path to S3 address, without having to store data locally

**Comment by Andy-P** [comment]
Just wanted to let you all know that I successfully built MXNet on AWS using an instance of the public cuda/cudnn GPU AMI above and am now running MXNet.jl through IJulia. 

Haven't tried S3 or EC2 GPU Clusters yet. Still just experimenting.

**Comment by pluskid** [comment]
@Andy-P Great! :D

**Comment by Andy-P** [comment]
Leaving this note for others that may view this thread. 

I created an AWS AMI for my presentation on 12/19 at Julia Tokyo no. 5. If you just want to kick the tires on MXNet.jl it is available at:
https://console.aws.amazon.com/ec2/v2/home?region=us-east-1#LaunchInstanceWizard:ami=ami-eda8fc87

It uses the most current versions of Cuda 7.0, MXNet and MXNet.jl as of 12/18. I did **not** enable S3, or USE_DIST_KVSTORE flag (i.e. no multi machine)
